packages/sdk/src/libs/clients/lsp/client.ts

```diff
@@ -0,0 +1,103 @@
+import { LongShortPair__factory, LongShortPair } from "@uma/core/contract-types/ethers";
+import type { SignerOrProvider, GetEventType } from "../..";
+import { Event } from "ethers";
+import { Balances } from "../../utils";
+
+export type Instance = LongShortPair;
+
+export type TokensCreated = GetEventType<Instance, "TokensCreated">;
+export type TokensRedeemed = GetEventType<Instance, "TokensRedeemed">;
+export type ContractExpired = GetEventType<Instance, "ContractExpired">;
+export type PositionSettled = GetEventType<Instance, "PositionSettled">;
+
+export function connect(address: string, provider: SignerOrProvider): Instance {
+  return LongShortPair__factory.connect(address, provider);
+}
+
+export interface EventState {
+  sponsors?: string[];
+  longs?: Balances;
+  shorts?: Balances;
+  collateral?: Balances;
+  expired?: boolean;
+  expiredBy?: string;
+}
+
+export function reduceEvents(state: EventState = {}, event: Event, index?: number): EventState {
+  switch (event.event) {
+    case "TokensCreated": {
+      const typedEvent = event as TokensCreated;
```

Comment:
just to understand correctly: what does this type enforce exactly?

Reply:
this just tells typescript that the generic event is a TokensCreated event. This will prevent us from accessing props on the event.args or other parts of the event which dont actually exist for this event type.